### PR TITLE
Move to C++11 & replace `pthread` with `std::thread`

### PR DESCRIPTION
**Pull request by clementperon**
Hi,

This MR try to remove pthread requiement on Windows by switching to C++ Threads introduce in C++11.

I keep pthread on Linux as we use the 'pthread_setaffinity_np()' and there is no equivalement with C++ threads.

This is untested!

### CODE REVIEW

**Comment by seladb** [comment]
> I keep pthread on Linux as we use the 'pthread_setaffinity_np()' and there is no equivalement with C++ threads.

@clementperon if we want we can replace `pthread` with `std::thread` and use `pthread_setaffinity_np` with `thread.native_handle()`, as described here: https://stackoverflow.com/questions/24645880/set-cpu-affinity-when-create-a-thread
That way we can probably remove `pthread` from all header files, including DPDK and PF_RING.

However, maybe we should do it in a separate PR? This PR has already changes in 41 files. Please let me know what you think

**Comment by seladb** [comment]
Also - in [`KniDevice.cpp`](https://github.com/seladb/PcapPlusPlus/blob/master/Pcap%2B%2B/src/KniDevice.cpp) I think we can replace `pthread` with `std::thread`, we don't use `pthread_setaffinity_np` there. But again - maybe we should do it in a separate PR?

**Comment by clementperon** [comment]
Agree just one thing we lost here is the mingw32 support?

**Comment by seladb** [comment]
> Agree just one thing we lost here is the mingw32 support?

Yes, MinGW doesn't support `std::thread`... only MinGW-w64

**Comment by seladb** [comment]
@clementperon can you run the tests and the example apps on as many OS's as you can?
We have the CI, but in this case it won't hurt to do some manual testing as well

File: Pcap++/src/PcapLiveDevice.cpp [hunk 1]
```diff
@@ -997,15 +950,8 @@ const std::vector<IPv4Address>& PcapLiveDevice::getDnsServers() const
 	return PcapLiveDeviceList::getInstance().getDnsServers();
 }
 
-ThreadStart PcapLiveDevice::getCaptureThreadStart()
-{
-	return &captureThreadMain;
-}
```
[the remark this answers is not in the document]
Comment (clementperon) [reply Pcap++/src/PcapLiveDevice.cpp:1003]: because there is no override symbol. So PcapRemoteDevice don't complain

**Comment by clementperon** [comment]
> @clementperon can you run the tests and the example apps on as many OS's as you can? We have the CI, but in this case it won't hurt to do some manual testing as well

I just have an Ubuntu 22.04 and Tests are fine

**Comment by seladb** [comment]
Ah ok, I'll test it more on Windows, macOS, FreeBSD (VM) and CentOS (VM)

**Pull request merged**